src/main/java/com/thebois/models/world/structures/AbstractStructure.java

```diff
@@ -61,4 +66,62 @@ public StructureType getType() {
         return structureType;
     }
 
+    @Override
+    public Collection<IItem> neededItems() {
+        // Make a copy of all IItems in collectionOfAllNeededItems
+        final Collection<IItem> itemDifference = new ArrayList<>(allNeededItems);
+        // Remove IItems that have been delivered
+        deliveredItems.forEach(itemDifference::remove);
+        // Return difference of the two collections
+        return itemDifference;
+    }
+
+    @Override
+    public boolean deliverItem(final IItem deliveredItem) {
```

Comment:
style

I prefer to prefix this type of method with `try`, so it would be `tryDeliverItem`. Since it can fail if the item is not needed, what the caller is actually doing is *trying* to deliver an item.